### DISCUSSION TITLE
GPS positioning in China appears to be far off

### DISCUSSION THREAD

**Discussion by alanchengh** [meshtastic/Discussions#25] [iOS]
Hello, we found that when using positioning in China, the displayed position is offset. Because Chinese maps use GCJ-02 coordinates. GCJ-02 coordinate system, also known as "Mars coordinate system", It is an original coordinate system created by my country's National Survey Bureau and is encrypted by WGS-84. In China, at least the GCJ-02 coordinate system must be used, or a coordinate system that is encrypted after GCJ-02 encryption and then encrypted, such as the Baidu coordinate system. Amap and Google both use the GCJ-02 coordinate system in China. It can be said that GCJ-02 is the most widely used coordinate system in China.

**Comment by hayschan** [comment]
Same for me.

**Comment by cezza168** [comment]
GCJ-02 Algorithmic Offset?
https://en.wikipedia.org/wiki/Restrictions_on_geographic_data_in_China

**Reply by hayschan** [reply]
That's right. The algorithm of converting the offset is quite easy though.

I have written an issue over here explaining the possible fix and related code snippet. 

https://github.com/meshtastic/Meshtastic-Apple/issues/981#issuecomment-2449132963
